### PR TITLE
cli: add --relative-path option

### LINKED ISSUE #3179: UnlockWallet relative to config instead of node execution dir
## Is your feature request related to a problem? Please describe.

The config file contains a section for `Consensus` configuration. The `UnlockWallet.Path` appears to be relative to the directory where the node is started from.  I'm currently facing a problem with specifying the wallet location in a setup where the node starting directory changes, but the configuration file + wallet could (and should) stay the same. 

The configuration file and wallet file have a tight relationship, but its implementation has the relationship between the node and the wallet file, not the configuration file. Take this folder that contains all the node needs to know
```
~/data/
├── consensus_wallet.json
├── notary_wallet.json
└── protocol.privnet.yml
```

With the great `--config-file` option introduced in https://github.com/nspcc-dev/neo-go/pull/3014 I would love to be able to run the node from any working directory by simply pointing it at the config e.g.
```shell
~/node/neo-go node --config-file ~/data/protocol.privnet.yml
```
or 
```shell
~/testnode/neo-go node --config-file ~/data/protocol.privnet.yml
```
should both work.

## Describe the solution you'd like

A way of  specifying the Consensus wallet path that is not relative to the node start directory. Can be a mutually exclusive `RelativePath` field, or a cli option like `--consensus-wallet-path` that is an absolute path

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Closes #3179

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
Please, change the PR name to something like `cli: add --relative-path option` or `Add --relative-path option to CLI`. See the https://github.com/nspcc-dev/.github/blob/master/git.md#commit-messages, the same pattern can be applied to the PR names.

**Review by AnnaShaleva — COMMENTED** [review]
Please, change the commit message. See the https://github.com/nspcc-dev/neo-go/pull/3206#issuecomment-1818999947, this comment is also relevant for the commit messages. And try to fit the message into one line (69 characters or so).

File: cli/smartcontract/testdata/rpcbindings/notifications/rpcbindings_guessed.out [hunk 1]
```diff
@@ -82,7 +82,6 @@ type LedgerWitnessRule struct {
 // Unnamed is a contract-specific unnamed type used by its methods.
 type Unnamed struct {
 	I *big.Int
-	B bool
```
Comment (roman-khimov) [inline cli/smartcontract/testdata/rpcbindings/notifications/rpcbindings_guessed.out:85]: Shouldn't be committed.

**Comment by roman-khimov** [comment]
Needs a rebase after recent merges.

File: cli/options/options.go [hunk 2]
```diff
@@ -73,6 +73,13 @@ var ConfigFile = cli.StringFlag{
 	Usage: "path to the node configuration file (overrides --config-path option)",
 }
 
+// RelativePath is a flag for commands that use node configuration and provide
+// a prefix to all relative paths in config files.
+var RelativePath = cli.StringFlag{
+	Name:  "relative-path",
+	Usage: "a prefix to all relative paths in config files",
```
Comment (AnnaShaleva) [inline cli/options/options.go:80]: s/`in config files`/`in the node configuration file`, just to match the description of other config-related options.

File: cli/server/server_test.go [hunk 3]
```diff
@@ -49,6 +49,16 @@ func TestGetConfigFromContext(t *testing.T) {
...
 	})
+	t.Run("relative-path", func(t *testing.T) {
+		set := flag.NewFlagSet("flagSet", flag.ExitOnError)
+		set.String("relative-path", "../../config", "")
+		set.Bool("testnet", true, "")
+		set.String("config-file", "../../config/protocol.testnet.yml", "")
+		ctx := cli.NewContext(cli.NewApp(), set, nil)
+		cfg, err := options.GetConfigFromContext(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "../../config/chains/testnet", cfg.ApplicationConfiguration.DBConfiguration.LevelDBOptions.DataDirectoryPath)
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:60]: Good, and also we can add the same check for absolute paths in `protocol.testnet.yml` (they should not be changed).

File: pkg/config/config.go [hunk 4]
```diff
@@ -58,14 +59,14 @@ func (c Config) Blockchain() Blockchain {
 
 // Load attempts to load the config from the given
 // path for the given netMode.
-func Load(path string, netMode netmode.Magic) (Config, error) {
+func Load(path string, netMode netmode.Magic, relativePath string) (Config, error) {
```
Comment (AnnaShaleva) [inline pkg/config/config.go:62]: Let's make the argument variadic, then you don't need to change all occurances of `Load` and `LoadFile`. E.g.  `Load(path string, netMode netmode.Magic, relativePath ...string)` and `LoadFile(configPath string, relativePath ...string)`. Also, the documentation should be updated.

File: pkg/core/blockchain_neotest_test.go [hunk 5]
```diff
@@ -1100,10 +1100,9 @@ func TestBlockchain_MPTDeleteNoKey(t *testing.T) {
 
 // Test that all default configurations are loadable.
 func TestConfig_LoadDefaultConfigs(t *testing.T) {
-	var prefixPath = filepath.Join("..", "..", "config")
 	check := func(t *testing.T, cfgFileSuffix any) {
-		cfgPath := filepath.Join(prefixPath, fmt.Sprintf("protocol.%s.yml", cfgFileSuffix))
-		_, err := config.LoadFile(cfgPath)
+		cfgPath := filepath.Join("..", "..", "config", fmt.Sprintf("protocol.%s.yml", cfgFileSuffix))
+		_, err := config.LoadFile(cfgPath, "")
```
Comment (AnnaShaleva) [inline pkg/core/blockchain_neotest_test.go:1105]: Useless change.

**Review by AnnaShaleva — COMMENTED** [review]
Looks better.

File: cli/options/options.go [hunk 6]
```diff
@@ -160,14 +167,15 @@ func GetRPCWithInvoker(gctx context.Context, ctx *cli.Context, signers []transac
 // returns an appropriate config.
 func GetConfigFromContext(ctx *cli.Context) (config.Config, error) {
 	var configFile = ctx.String("config-file")
+	var relativePath = ctx.String("relative-path")
```
Comment (AnnaShaleva) [inline cli/options/options.go:170]: These two variables may be moved under a single `var` definition.

File: pkg/config/config.go [hunk 7]
```diff
@@ -57,15 +58,17 @@ func (c Config) Blockchain() Blockchain {
...
 
 // Load attempts to load the config from the given
-// path for the given netMode.
-func Load(path string, netMode netmode.Magic) (Config, error) {
+// path for the given netMode. If relativePath is not empty, relative paths in the
+// config will be updated based on the provided relative path.
+func Load(path string, netMode netmode.Magic, relativePath ...string) (Config, error) {
 	configPath := fmt.Sprintf("%s/protocol.%s.yml", path, netMode)
-	return LoadFile(configPath)
+	return LoadFile(configPath, relativePath[0])
```
Comment (AnnaShaleva) [inline pkg/config/config.go:65]: s/`relativePath[0]`/`relativePath...`

Consider the case when no `relativePath` is provided. Then panic occurs.

**Review by AnnaShaleva — COMMENTED** [review]
Minor issues are left, otherwise LGTM.

File: cli/options/options.go (hunk 6 above)
Comment (AnnaShaleva) [reply cli/options/options.go:170]: Not resolved.

File: cli/options/options.go [hunk 8]
```diff
@@ -160,14 +167,18 @@ func GetRPCWithInvoker(gctx context.Context, ctx *cli.Context, signers []transac
 // returns an appropriate config.
 func GetConfigFromContext(ctx *cli.Context) (config.Config, error) {
 	var configFile = ctx.String("config-file")
+	var relativePath = ""
+	if argrPath := ctx.String("relative-path"); argrPath != "" {
+		relativePath = argrPath
+	}
```
Comment (AnnaShaleva) [inline cli/options/options.go:173]: The previous variant was correct, we don't need this extra `if`. Just use the following:
```suggestion
var (
	configFile = ctx.String("config-file")
	relativePath = ctx.String("relative-path")
)
```

File: cli/vm/cli_test.go [hunk 9]
```diff
@@ -140,7 +142,7 @@ func newTestVMClIWithState(t *testing.T) *executor {
 
 	// After that create CLI backed by created chain.
 	configPath := "../../config/protocol.unit_testnet.yml"
-	cfg, err := config.LoadFile(configPath)
+	cfg, err := config.LoadFile(configPath, "")
```
Comment (AnnaShaleva) [inline cli/vm/cli_test.go:145]: Useless change.

Comment (AliceInHunterland) [reply cli/vm/cli_test.go]: removed

File: cli/server/server_test.go [hunk 10]
```diff
@@ -233,7 +245,7 @@ func TestRestoreDB(t *testing.T) {
 		badCfgDir := t.TempDir()
 		logfile := filepath.Join(badCfgDir, "logdir")
 		require.NoError(t, os.WriteFile(logfile, []byte{1, 2, 3}, os.ModePerm))
-		cfg, err := config.LoadFile(filepath.Join(goodCfg, "protocol.privnet.yml"))
+		cfg, err := config.LoadFile(filepath.Join(goodCfg, "protocol.privnet.yml"), "")
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:248]: Useless change.

File: cli/server/server_test.go [hunk 11]
```diff
@@ -249,7 +261,7 @@ func TestRestoreDB(t *testing.T) {
 	})
 	t.Run("invalid bc config", func(t *testing.T) {
 		badCfgDir := t.TempDir()
-		cfg, err := config.LoadFile(filepath.Join(goodCfg, "protocol.privnet.yml"))
+		cfg, err := config.LoadFile(filepath.Join(goodCfg, "protocol.privnet.yml"), "")
```
Comment (AnnaShaleva) [inline cli/server/server_test.go:264]: Useless change.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]
Good.

**Pull request merged**